sandboxes/andromeda/consumer.cannonfile.toml

```diff
@@ -0,0 +1,288 @@
+depends = ["invoke.delegate_10k_synth_usdc_to_pool"]
+
+[invoke.approve_20k_usdc_for_SpotMarketProxy_for_wrapping]
+target = ["andromeda_sandbox.usdc_token.MintableToken"]
+from = "<%= settings.test_user %>"
+func = "approve"
+args = [
+    "<%= imports.andromeda_sandbox.imports.spot_factory.contracts.SpotMarketProxy.address %>",
+    "<%= parseEther(String(20_000)) %>",
+]
+depends = ["invoke.mint_20k_usdc_for_perps"]
+
+[invoke.wrap_20k_usdc_into_synth_usdc]
+target = ["andromeda_sandbox.spot_factory.SpotMarketProxy"]
+from = "<%= settings.test_user %>"
+func = "wrap"
+args = [
+    # uint128 marketId
+    "<%= imports.andromeda_sandbox.extras.synth_usdc_market_id %>",
+    # uint256 wrapAmount
+    "<%= parseEther(String(20_000)) %>",
+    # uint256 minAmountReceived (0% slippage!)
+    "<%= parseEther(String(20_000)) %>"
+]
+depends = ["invoke.approve_20k_usdc_for_SpotMarketProxy_for_wrapping"]
+
+[invoke.atomic_sell_20k_synth_usdc_for_20k_susd]
+target = ["andromeda_sandbox.spot_factory.SpotMarketProxy"]
+from = "<%= settings.test_user %>"
+func = "sell"
+args = [
+    # uint128 marketId
+    "<%= imports.andromeda_sandbox.extras.synth_usdc_market_id %>",
+    # uint256 synthAmount
+    "<%= parseEther(String(20_000)) %>",
+    # uint256 minUsdAmount (0% slippage!)
+    "<%= parseEther(String(20_000)) %>",
+    # address referrer
+    "<%= AddressZero %>"
+]
+depends = ["invoke.wrap_20k_usdc_into_synth_usdc"]
+
+#
+#
+#
+# Depositing sUSD into the Perps market
+#
+#
+#
+[invoke.create_perps_account]
+target = ["andromeda_sandbox.perps_factory.PerpsMarketProxy"]
+from = "<%= settings.test_user %>"
+func = "createAccount(uint128)"
+args = ["<%= settings.test_user_perps_account_id %>"]
+extra.test_user_perps_account_id.event = "AccountCreated"
+extra.test_user_perps_account_id.arg = 0
+depends = [
+    # We want to ensure pool has some liquidity before test user start interacting with perps market, so adding this dependency step
+    "invoke.delegate_10k_synth_usdc_to_pool",
+    # We want to ensure user has sUSD available in their wallet too
+    "invoke.atomic_sell_20k_synth_usdc_for_20k_susd"
+]
+
+[invoke.approve_1k_susd_for_PerpsMarketProxy_to_trade]
+target = ["andromeda_sandbox.synthetix.USDProxy"]
+from = "<%= settings.test_user %>"
+func = "approve"
+args = [
+    "<%= imports.andromeda_sandbox.imports.perps_factory.contracts.PerpsMarketProxy.address %>",
+    "<%= parseEther(String(1_000)) %>",
+]
+depends = ["invoke.create_perps_account"]
+
+[invoke.deposit_1k_susd_to_perps]
+target = ["andromeda_sandbox.perps_factory.PerpsMarketProxy"]
+from = "<%= settings.test_user %>"
+func = "modifyCollateral"
+args = [
+    "<%= extras.test_user_perps_account_id %>",
+    0, # 0 is a special market id for susd
+    "<%= parseEther(String(1_000)) %>",
+]
+depends = ["invoke.approve_1k_susd_for_PerpsMarketProxy_to_trade"]
+
+#
+#
+#
+# Withdrawing 300 sUSD from perps to get 300 USDC back into the wallet
```

Comment:
Is the purpose of this to test things? Or why dont we just deposit $700 instead of depositing $1k and then withdrawing

Reply:
Yes. This is a `consumer` of sandbox. That is part of our "documentation" + automated testing. We will have auto-tests running for this. to ensure any updates to contracts do not break the sandbox operations. And if they do - means we have breaking changes that need to be communicated to integrators.